apps/emqx_durable_storage/src/emqx_ds_storage_bitfield_lts.erl

```diff
@@ -412,7 +514,117 @@ traverse_interval(ITHandle, Filter, Cutoff, It, Acc, N) ->
             {0, It, Acc}
     end.
 
--spec check_message(emqx_ds:time(), iterator(), emqx_types:message()) ->
+delete_next_loop(
+    _DB, _CF, _ITHandle, _KeyMapper, _Filter, _Cutoff, It, AccDel, AccIter, _Selector, 0
+) ->
+    {ok, It, AccDel, AccIter};
+delete_next_loop(DB, CF, ITHandle, KeyMapper, Filter, Cutoff, It0, AccDel0, AccIter0, Selector, N0) ->
+    inc_counter(),
+    #{?tag := ?DELETE_IT, ?last_seen_key := Key0} = It0,
+    case emqx_ds_bitmask_keymapper:bin_increment(Filter, Key0) of
+        overflow ->
+            {ok, It0, AccDel0, AccIter0};
+        Key1 ->
+            %% assert
+            true = Key1 > Key0,
+            case rocksdb:iterator_move(ITHandle, {seek, Key1}) of
+                {ok, Key, Val} ->
+                    {N, It, AccDel, AccIter} =
+                        delete_traverse_interval(
```

Comment:
Nit: maybe it also makes sense to hide some args inside some kind of a "context"... In fdb replica of the algo, I already had to because the mental burden became too heavy 😅